apps/root/src/common/components/wallet-selector/index.tsx

```diff
@@ -28,25 +28,48 @@ const StyledMenuItem = styled(MenuItem)`
   gap 5px; 
 `;
 
-interface BaseProps {}
-
-interface WithAllWallets {
-  allowAllWalletsOption: boolean;
+type WithAllWalletsOption = {
+  allowAllWalletsOption: true;
+  setSelectionAsActive?: never;
   onSelectWalletOption: (newWallet: string) => void;
   selectedWalletOption: string;
-}
+};
 
-interface WithoutAllWallets {
+type WithSetActiveWalletTrue = {
   allowAllWalletsOption?: never;
+  setSelectionAsActive: true;
   onSelectWalletOption?: never;
   selectedWalletOption?: never;
-}
+};
+
+type WithSetActiveWalletFalse = {
+  allowAllWalletsOption?: never;
```

Comment:
if you don't want this props, you can just not write them, and then do an join on typescript `|`

Reply:
Or is that a problem with props?

Reply:
Every prop should be defined across all union types or it would not be recognized by the component.
https://stackoverflow.com/questions/58630750/react-typescript-component-with-two-different-prop-interfaces

Reply:
Sooo nice!!!! tytyty